bittide-instances/src/Clash/Shake/Vivado.hs

```diff
@@ -94,15 +95,18 @@ mkSynthesisTcl ::
   Bool ->
   -- | Part to synthesize for. E.g., 'xcku040-ffva1156-2-e'.
   String ->
+  -- | List of filepaths to XDC files
+  [FilePath] ->
   -- | Manifests of which the first is the top-level to synthesize
   LocatedManifest ->
   -- | Rendered TCL
   IO String
-mkSynthesisTcl outputDir outOfContext part manifest@LocatedManifest{lmManifest} = do
+mkSynthesisTcl outputDir outOfContext part constraints manifest@LocatedManifest{lmManifest} = do
   baseTcl <- mkBaseTcl outputDir manifest
   pure $ baseTcl <> "\n" <> [__i|
     set_msg_config -severity {CRITICAL WARNING} -new_severity ERROR
 
+    #{constraintsString}
```

Comment:
What does this do?

Reply:
It adds the following line for each given filepath to an XDC file, which I added as a new input to `mkSynthesisTcl` (see the `where` clause)
`read_xdc [filepath_to_pinmapping]`